microraiden/microraiden/test/utils/client.py

```diff
@@ -26,3 +28,13 @@ def close_all_channels_cooperatively(
                 if c.state != Channel.State.closed and c.receiver == receiver_addr]
     for channel in channels:
         close_channel_cooperatively(channel, privkey_receiver, contract_address, balance)
+
+
+def patch_on_http_response(default_http_client: DefaultHTTPClient, abort_on=[]):
+    def patched(self, resource, response):
+        self.last_response = response
+        return (response.status_code not in abort_on)
+    default_http_client.on_http_response = types.MethodType(
```

Comment:
Just a note (no changes needed):

`pytest` has the `monkeypatch` fixture (of type `_pytest.monkeypatch.Monkeypatch`) which would be a nice alternative to these `types.MethodType(...)` calls. For example I use this to mock `click` prompt inputs:

```python
def mock_prompt(monkeypatch: MonkeyPatch, input_sequence: Iterator[str]):
    def patched_prompt(*args, **kwargs):
        value = next(input_sequence)
        print('{}: {}'.format(args[0], value))
        return value

    monkeypatch.setattr(click, 'prompt', patched_prompt)
    monkeypatch.setattr(click, 'confirm', patched_prompt)
```

It automatically cleans the monkeypatching up after the test and allows to patch built-in methods too.

Reply:
I've just tried that and it doesn't seem to be working with instances.

```python
def patched(self, , response):
          self.last_response = resp
          return False
monkeypatch.setattr(default_http_client, 'on_http_response', patched)
```

fails with 
`TypeError: patched() missing 1 required positional argument: 'response'`

Reply:
Ok nevermind that comment then. It only really works with built-in modules I guess. With custom classes it gets ugly because you need to specify the import path.